hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metrics/MetricsReporterFactory.java

```diff
@@ -56,6 +56,9 @@ public static MetricsReporter createReporter(HoodieWriteConfig config, MetricReg
 
     MetricsReporterType type = config.getMetricsReporterType();
     MetricsReporter reporter = null;
+    if (type == null) {
+      return reporter;
```

Comment:
this can return `null` reporter and we have to be extra careful at the caller, e.g. when it is called to add to the list in `addAdditionalMetricsExporters` it will throw and exception. It's better to handle null here itself.

Reply:
If type is null, there is no default reporter created. Type is used to determine whether reporter needs to be created or not today. So I am not throwing an exception here but I am converting return type to Option for this function.